Bugzilla: 1056711
Upstream-status: Submitted for 3.15

This fixes the issue described here:
https://bugzilla.redhat.com/show_bug.cgi?id=1056711#c5

net-next commit 971a351ccbbd2b6eef136a2221da0b80aca50906
Author: Jiri Pirko <jiri@resnulli.us>
Date:   Tue Dec 10 13:56:29 2013 +0100

    ipv6 addrconf: revert /proc/net/if_inet6 ifa_flag format
    
    Turned out that applications like ifconfig do not handle the change.
    So revert ifa_flag format back to 2-letter hex value.
    
    Introduced by:
    commit 479840ffdbe4242e8a25349218c8e0859223aa35
        "ipv6 addrconf: extend ifa_flags to u32"
    
Signed-off-by: Jiri Pirko <jpirko@redhat.com>

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index be4dbbd..3c3425e 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -3367,12 +3367,12 @@ static void if6_seq_stop(struct seq_file *seq, void *v)
 static int if6_seq_show(struct seq_file *seq, void *v)
 {
 	struct inet6_ifaddr *ifp = (struct inet6_ifaddr *)v;
-	seq_printf(seq, "%pi6 %02x %02x %02x %03x %8s\n",
+	seq_printf(seq, "%pi6 %02x %02x %02x %02x %8s\n",
 		   &ifp->addr,
 		   ifp->idev->dev->ifindex,
 		   ifp->prefix_len,
 		   ifp->scope,
-		   ifp->flags,
+		   (u8) ifp->flags,
 		   ifp->idev->dev->name);
 	return 0;
 }

_______________________________________________
kernel mailing list
kernel@lists.fedoraproject.org
https://admin.fedoraproject.org/mailman/listinfo/kernel
